src/dq_suite/common.py

```diff
@@ -54,7 +55,54 @@ def __getitem__(self, key) -> str | Dict[str, Any] | int | None:
         raise KeyError(key)
 
 
-RulesList = list[Rule]  # a list of DQ rules
+@dataclass
+class GeoRule(Rule):
+    """A specialized rule for geospatial validation."""
+    rule_type: Literal["geo"] = "geo"
```

Comment:
Is this really necessary? The name of the dataclass already says it's geodata - so if you need to check for the value of this `rule_type` attribute, you'd better check for an `isinstance` of the `GeoRule` class...